File: golang/version1/main.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"strings"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		numero := r.URL.Query().Get("n")

		xml := fmt.Sprintf(`<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
      <ubiNum>%s</ubiNum>
    </NumberToWords>
  </soap:Body>
</soap:Envelope>`, numero)

		req, err := http.NewRequest(
			"POST",
			"https://www.dataaccess.com/webservicesserver/NumberConversion.wso",
			strings.NewReader(xml),
		)

		if err != nil {
			fmt.Fprint(w, "Error: ", err)
			return
		}

		req.Header.Set("Content-Type", "text/xml; charset=utf-8")
		req.Header.Set("SOAPAction", "http://www.dataaccess.com/webservicesserver/NumberToWords")

		cliente := &http.Client{}
		resp, err := cliente.Do(req)

		if err != nil {
			fmt.Fprint(w, "Error: ", err)
			return
		}

		defer resp.Body.Close()

		body, _ := io.ReadAll(resp.Body)
		respuesta := string(body)

		inicio := strings.Index(respuesta, "<m:NumberToWordsResult>")
		fin := strings.Index(respuesta, "</m:NumberToWordsResult>")

		if inicio == -1 || fin == -1 {
			inicio = strings.Index(respuesta, "<NumberToWordsResult>")
			fin = strings.Index(respuesta, "</NumberToWordsResult>")

			if inicio == -1 || fin == -1 {
				fmt.Fprint(w, "No se encontró resultado")
				return
			}

			inicio += len("<NumberToWordsResult>")
		} else {
			inicio += len("<m:NumberToWordsResult>")
		}

		resultado := strings.TrimSpace(respuesta[inicio:fin])
		fmt.Fprint(w, resultado)
	})

	fmt.Println("Servidor Go version1 en http://localhost:8090/?n=10")
	http.ListenAndServe(":8090", nil)
}

File: ruby/version1/main.rb
require 'webrick'
require 'net/http'

server = WEBrick::HTTPServer.new(
  Port: 8100
)

server.mount_proc '/' do |req, res|

  numero = req.query['n']

  xml = <<~XML
  <?xml version="1.0" encoding="utf-8"?>
  <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
    <soap:Body>
      <NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
        <ubiNum>#{numero}</ubiNum>
      </NumberToWords>
    </soap:Body>
  </soap:Envelope>
  XML

  uri = URI(
    'https://www.dataaccess.com/webservicesserver/NumberConversion.wso'
  )

  respuesta = Net::HTTP.post(
    uri,
    xml,
    {
      'Content-Type' => 'text/xml; charset=utf-8',
      'SOAPAction' => 'http://www.dataaccess.com/webservicesserver/NumberToWords'
    }
  )

  cuerpo = respuesta.body

  resultado = cuerpo[
    /<m:NumberToWordsResult>(.*?)<\/m:NumberToWordsResult>/m,
    1
  ]

  if resultado.nil?

    resultado = cuerpo[
      /<NumberToWordsResult>(.*?)<\/NumberToWordsResult>/m,
      1
    ]

  end

  res.body = resultado
end

trap('INT') do
  server.shutdown
end

puts 'Servidor Ruby version1 en http://localhost:8100/?n=10'

server.start